### ISSUE TITLE
[bitnami/moodle] TLS/SSL Error when trying to start a Moodle container using an external MySQL Database with self-signed certificates

### ISSUE THREAD

**Issue by erickcomp** [bitnami/containers#70006]
### Name and Version

bitnami/moodle:4.2.8-debian-12-r4 

### What architecture are you using?

amd64

### What steps will reproduce the bug?

When I updated the image in my staging environment, I got an error starting Moodle:
```
ERROR 2026 (HY000): TLS/SSL error: self-signed certificate in certificate chain
moodle 16:51:12.68 ERROR ==> Could not connect to the database
```
Steps to reproduce:
1 - Update the image from "bitnami/moodle:4.2.8-debian-12-r3 " to "bitnami/moodle:4.2.8-debian-12-r4"
2 - Use a non-dockerized database server with self-signed certificate

### What is the expected behavior?

Start the moodle service without SSL/TLS errors when connecting to the database server

### What do you see instead?

ERROR 2026 (HY000): TLS/SSL error: self-signed certificate in certificate chain 
moodle 16:51:12.68 ERROR ==> Could not connect to the database

### Additional information

This new mysql client seems to be the culprit:
mysql-client-11.4.2-0-linux-${OS_ARCH}-debian-12

**Comment by javsalgar** [comment]
Hi,

Did the error happen with the previous image? Just to confirm there is a regression between the two images

**Comment by erickcomp** [comment]
Hi, @javsalgar . Yes, I forced bitnami/moodle:4.2.8-debian-12-r3 on my Dockerfile and it's working normally.
If I use bitnami/moodle:4.2.8-debian-12-r4 or just bitnami/moodle:4.2.8, it fails with the error described here.

**Comment by beertje44** [comment]
Finally got it working again: somehow on both ends the env vars for SSL certs don't seem to work so it uses the default mysql server-crt.pem which is self signed, so I added the ca.pem to the debian key store of the bitnami wordpress-nginx image. No still no game, but different error 👎 In the end I had to create default mysql certs and ca with a CN that matches my mysql container internal hostname.

Shouldn't be this hard!

**Comment by juan131** [comment]
Hi everyone!

As you can see in the commit below, MySQL client was bumped from `11.3.2` to `11.4.2` version on that image revision:

- https://github.com/bitnami/containers/commit/99ae6eba622b0a19d3a5b465cb968a03bc226e1a

It seems this version introduces changes in the TLS encryption, see:

- https://mariadb.com/kb/en/mariadb-11-4-2-release-notes/#encryption-tls-ssl
- https://jira.mariadb.org/browse/MDEV-33592

**Comment by designamx** [comment]
I am having the same issue with version docker ` bitnami/moodle:4.4.1`

**Comment by erickcomp** [comment]
> Hi everyone!
> 
> As you can see in the commit below, MySQL client was bumped from `11.3.2` to `11.4.2` version on that image revision:
> 
>     * [99ae6eb](https://github.com/bitnami/containers/commit/99ae6eba622b0a19d3a5b465cb968a03bc226e1a)
> 
> 
> It seems this version introduces changes in the TLS encryption, see:
> 
>     * https://mariadb.com/kb/en/mariadb-11-4-2-release-notes/#encryption-tls-ssl
> 
>     * https://jira.mariadb.org/browse/MDEV-33592

As I imagined...
So, do we have any options around this?

Thanks

**Comment by juan131** [comment]
Hi @erickcomp 

Have you tried using self-signed certificates which CN matches the hostname you're using to access Moodle?

**Comment by erickcomp** [comment]
Hey, @juan131. I haven't tried anything yet. I rolled back the version to the r3 to use the old client until I got time to sit down and check this. One more thing: I always get the message about the MySQL program/client deprecation, not Maria DB. I wonder if the modifications on MariaDB were applied to the MySQL client as well.

**Comment by designamx** [comment]
since I was doing a fresh installation I decided to get rid of mysql server and install mariadb, that did the trick for me

**Comment by juan131** [comment]
Hi @erickcomp 

The `mysql-client` included in the Bitnami Moodle container is actually built from MariaDB source code.

**Comment by juan131** [comment]
The same bug is being discussed at https://github.com/bitnami/containers/issues/70392

**Comment by gongomgra** [comment]
Hi everyone,

We have just released a new revision of our Moodle image including the fix for this issue. 

```
docker pull bitnami/moodle:4.2.8-debian-12-r6
```

You can check the changes included in this new image at https://github.com/bitnami/containers/commit/1a3a8a78acb6213062791cd982733e51bc051b49

**Comment by XaratosS** [comment]
I will give it a look later. Does it also fix the issue with the latest moodle version? 
As soon as I tested it, I will give you guys feedback.

**Comment by hypery2k** [comment]
with `docker.io/bitnami/moodle:4.4.2-debian-12-r0` the error is resolved

**Comment by erickcomp** [comment]
I'll take a look in the upcoming days. Anyway, thank you!

**Comment by juan131** [comment]
Solved on `bitnami/moodle:4.2.8-debian-12-r6` as described at https://github.com/bitnami/containers/issues/70006#issuecomment-2265632821
